tests/agent/test_agent_config.py

```diff
@@ -3,71 +3,150 @@
+        {"hosts": hosts, "username": username, "password": password}, None
+    )
 
-    source_mock.fields = fields_container
-    source_mock.__getitem__.side_effect = fields_container.__getitem__
-
-    assert config_wrapper.try_update(source_mock) is True
+    assert config_wrapper.try_update(unit_mock) is True
     assert config_wrapper.get()["elasticsearch"]["host"] == hosts[0]
     assert config_wrapper.get()["elasticsearch"]["username"] == username
     assert config_wrapper.get()["elasticsearch"]["password"] == password
+
+
+def test_try_update_multiple_times_does_not_reset_config_values():
+    hosts = ["https://localhost:9200"]
+    api_key = "lemme_in"
+
+    log_level = "DEBUG"
+    config_wrapper = ConnectorsAgentConfigurationWrapper()
+
+    # First unit comes with elasticsearch data
+    first_unit_mock = prepare_unit_mock({"hosts": hosts, "api_key": api_key}, None)
+
+    # Second unit comes only with a log_level
+    second_unit_mock = prepare_unit_mock({}, log_level)
+
+    assert config_wrapper.try_update(first_unit_mock) is True
+    assert config_wrapper.try_update(second_unit_mock) is True
+
+    assert config_wrapper.get()["elasticsearch"]["host"] == hosts[0]
+    assert config_wrapper.get()["elasticsearch"]["api_key"] == api_key
+    assert config_wrapper.get()["service"]["log_level"] == log_level
+
+
+def test_config_changed_when_new_variables_are_passed():
+    hosts = ["https://localhost:9200"]
+    api_key = "lemme_in_lalala"
+
+    new_config = {"elasticsearch": {"hosts": hosts, "api_key": api_key}}
+
+    config_wrapper = ConnectorsAgentConfigurationWrapper()
+
+    assert config_wrapper.config_changed(new_config) is True
+
+
+def test_config_changed_when_elasticsearch_config_changed():
+    hosts = ["https://localhost:9200"]
+    api_key = "lemme_in_lalala"
+
+    starting_config = {
+        "elasticsearch": {
+            "hosts": hosts,
+            "username": "elastic",
+            "password": "hey-im-a-password",
+        }
+    }
+    new_config = {"elasticsearch": {"hosts": hosts, "api_key": api_key}}
+
+    config_wrapper = ConnectorsAgentConfigurationWrapper()
+    config_wrapper.try_update(prepare_unit_mock(starting_config, None))
+
+    assert config_wrapper.config_changed(new_config) is True
+
+
+def test_config_changed_when_elasticsearch_config_did_not_change():
+    hosts = ["https://localhost:9200"]
+    api_key = "lemme_in_lalala"
+
+    new_config = {"elasticsearch": {"hosts": hosts, "api_key": api_key}}
+
+    config_wrapper = ConnectorsAgentConfigurationWrapper()
+    config_wrapper.try_update(prepare_unit_mock(new_config, None))
+
+    assert config_wrapper.config_changed(new_config) is True
+
+
+def test_config_changed_when_log_level_config_changed():
+    config_wrapper = ConnectorsAgentConfigurationWrapper()
+    config_wrapper.try_update(prepare_unit_mock({}, "INFO"))
+
+    new_config = {"service": {"log_level": "DEBUG"}}
+
+    assert config_wrapper.config_changed(new_config) is True
```

Comment:
Might be interesting to modify this test or add another to actually assert that the log level does change. That you get new log statements after the new config is passed.

Reply:
Added more comprehensive test in https://github.com/elastic/connectors/pull/2809/commits/9d14acd7f470f203cf49d0ebeed96e9ea3d0b2b2 it tests if log level from unit (from agent protocol) correctly gets propagated to the resulting configuration